ocsp/src/main/java/io/nats/ocsp/OcspExample.java

```diff
@@ -74,39 +66,28 @@ public static SSLContext getSiloedContext()
-        // that pem file. You likely will have to figure out how to load your own certificates
-        // Then add the certificates to a Key Store. You will likely have a different way to make or
-        // load your own Key Store
-        List<X509Certificate> certificates = loadCertificates(ocspCaCertPath());
-        KeyStore keyStore = KeyStore.getInstance(KEYSTORE_TYPE);
-        keyStore.load(null, KEY_STORE_PASSWORD.toCharArray());
-        for (X509Certificate certificate : certificates) {
-            String alias = certificate.getSubjectX500Principal().getName();
-            keyStore.setCertificateEntry(alias, certificate);
-        }
+        // The example ca cert and key is in a pem file. This PemUtils makes a Trust Manager from it.
+        // You likely will have to figure out how to make your own Trust Manager
+        X509ExtendedTrustManager trustManager =
+                PemUtils.loadTrustMaterial(ocspCaCertPath());
 
         // The PKIXRevocationChecker is the class that does the work of checking the revocation
         CertPathBuilder cpb = CertPathBuilder.getInstance("PKIX");
         PKIXRevocationChecker rc = (PKIXRevocationChecker)cpb.getRevocationChecker();
         rc.setOptions(EnumSet.of(PKIXRevocationChecker.Option.NO_FALLBACK));
 
-        // PKIXBuilderParameters uses the Key Store and the Revocation Checker
-        PKIXBuilderParameters pkixParams = new PKIXBuilderParameters(keyStore, new X509CertSelector());
-        pkixParams.addCertPathChecker(rc);
-
-        // The CertPathTrustManagerParameters is made from the PKIX Builder parameters
-        CertPathTrustManagerParameters certPathTrustManagerParameters = new CertPathTrustManagerParameters(pkixParams);
-
-        // The Trust Manager Factory is made from the CertPathTrustManagerParameters
-        TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-        tmf.init(certPathTrustManagerParameters);
-
-        // Construct and initialize the sslContext
-        SSLContext sslContext = SSLContext.getInstance("TLSv1.2");
-        sslContext.init(new KeyManager[]{keyManager}, tmf.getTrustManagers(), new SecureRandom());
-
-        return sslContext;
+        return SSLFactory.builder()
+                .withIdentityMaterial(keyManager)
+                .withTrustMaterial(trustManager, trustStore -> {
+                    // PKIXBuilderParameters uses the Key Store and the Revocation Checker
+                    PKIXBuilderParameters pkixParams = new PKIXBuilderParameters(trustStore, new X509CertSelector());
+                    pkixParams.addCertPathChecker(rc);
+
+                    // The CertPathTrustManagerParameters is made from the PKIX Builder parameters
+                    return new CertPathTrustManagerParameters(pkixParams);
+                })
+                .build()
+                .getSslContext();
```

Comment:
So it's import to demonstrate the step by step building of the Trust Manager Factory as that is what I'm trying to show with the example.